test/common/stats/tag_extractor_impl_test.cc

```diff
@@ -106,10 +106,10 @@ class DefaultTagRegexTester {
    * assuming we don't care about tag-order. This is in large part correct by design because
    * stat_name is not mutated until all the extraction is done.
    * @param metric_name std::string a name of Stats::Metric (Counter, Gauge, Histogram).
-   * @param tags std::vector<Tag>& a set of Stats::Tag.
+   * @param tags TagVector& a set of Stats::Tag.
    * @return std::string the metric_name with tags removed.
    */
-  std::string produceTagsReverse(const std::string& metric_name, std::vector<Tag>& tags) const {
+  std::string produceTagsReverse(const std::string& metric_name, TagVector& tags) const {
```

Comment:
can you a drop a TODO here and `produceTags()` to skip the creation of the string-based tag-vector in this API and just go right to StatName? The regex code delivers string-views so we don't have to make std::string copies of them now that we have APIs that are ready for the StatName versions.

No need to do that whole refactor in this PR but might as well queue that up with a TODO. You can assign me.